js/customize-snapshots-admin.js

```diff
@@ -0,0 +1,73 @@
+( function( $ ) {
+	'use strict';
+
+	$( function() {
+		var $link, linkText, linkActions, dataSlug, initializeLink;
+
+		$link = $( '.snapshot-toggle-setting-removal' );
+		linkText = [ 'Remove setting', 'Restore setting' ];
+		linkActions = [ 'remove', 'restore' ];
+		dataSlug = 'cs-action';
+
+		initializeLink = function() {
+			$link.text( linkText[ 0 ] )
+				.data( dataSlug, linkActions[ 0 ] );
+		};
+
+		initializeLink();
+
+		$link.on( 'click', function( event ) {
+			var $clickedLink, $settingDisplay, clickedLinkAction, settingId;
+
+			event.preventDefault();
+
+			$clickedLink = $( this );
+			$settingDisplay = $( this ).parents( 'details' );
+			clickedLinkAction = $( this ).data( dataSlug );
+			settingId = $( this ).attr( 'id' );
+
+			this.isLinkSetToRemoveSetting = function() {
+				return linkActions[ 0 ] === clickedLinkAction;
+			};
+
+			this.hideSettingAndChangeLinkText = function() {
+				$clickedLink.text( linkText[ 1 ] )
+					.data( dataSlug, linkActions[ 1 ] )
+					.after( this.constructHiddenInputWithValue() );
+				$settingDisplay.removeAttr( 'open' )
+					.addClass( 'snapshot-setting-removed' );
+			};
+
+			this.constructHiddenInputWithValue = function() {
+				return $( '<input>' ).attr( {
+					'name': 'customize_snapshot_remove_settings[]',
+					'type': 'hidden'
+				})
+				.val( settingId );
+			};
+
+			this.isLinkSetToRestoreSetting = function() {
+				return linkActions[ 1 ] === clickedLinkAction;
+			};
+
+			this.showSettingAndChangeLinkText = function() {
+				$clickedLink.text( linkText[ 0 ] )
+					.data( dataSlug, linkActions[ 0 ] );
+				this.removeHiddenInputWithValue();
+				$settingDisplay.removeClass( 'snapshot-setting-removed' );
+			};
+
+			this.removeHiddenInputWithValue = function() {
+				$( 'input[value="' + settingId + '"]' ).remove();
```

Comment:
That's a good point, @westonruter. This [commit](https://github.com/xwp/wp-customize-snapshots/pull/84/commits/83ad0e9ee2983c6fc7e35c6c55836e9c4242f3f0#diff-19c3ccdc306cf3ee7efd933155ccae35R73) ensures that the `name` is `customize_snapshot_remove_settings[]`.